tstate/tstate_view.go

```diff
@@ -213,8 +222,8 @@ func (ts *TStateView) isUnchanged(ctx context.Context, key string, nval []byte,
 // Insert allocates and writes (or just writes) a new key to [tstate]. If this
 // action returns the value of [key] to the parent view, it reverts any pending changes.
 func (ts *TStateView) Insert(ctx context.Context, key []byte, value []byte) error {
-	// Inserting requires a Write Permissions, so we pass state.Write
-	if !ts.checkScope(ctx, key, state.Write) {
+	// To insert, the key needs to have Allocate or Write permissions, or both
+	if !ts.checkScope(ctx, key, state.Allocate, state.Write) {
```

Comment:
You can just check this using the `&` operator?

if `desired & provided == desired`